src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -0,0 +1,172 @@
+        try {
+            byte[] fileBytes = Files.readAllBytes(path);
+            return Base64.getEncoder().encodeToString(fileBytes);
+        }
+        catch (IOException e) {
+            throw new IrisInternalPyrisErrorException(e.getMessage());
+        }
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachmentForUpdate(AttachmentUnit attachmentUnit) {
+        int lectureUnitId = attachmentUnit.hashCode();
+        String lectureUnitName = attachmentUnit.getName();
+        int lectureId = attachmentUnit.getLecture().hashCode();
+        String lectureTitle = attachmentUnit.getLecture().getTitle();
+        int courseId = attachmentUnit.getLecture().getCourse().hashCode();
+        String courseTitle = attachmentUnit.getLecture().getCourse().getTitle();
+        String courseDescription = attachmentUnit.getLecture().getCourse().getDescription() == null ? "" : attachmentUnit.getLecture().getCourse().getDescription();
+        String base64EncodedPdf = attachmentToBase64(attachmentUnit);
+        return new PyrisLectureUnitWebhookDTO(true, artemisBaseUrl, base64EncodedPdf, lectureUnitId, lectureUnitName, lectureId, lectureTitle, courseId, courseTitle,
+                courseDescription);
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachmentForDeletion(AttachmentUnit attachmentUnit) {
+        int lectureUnitId = attachmentUnit.hashCode();
+        int lectureId = attachmentUnit.getLecture().hashCode();
+        int courseId = attachmentUnit.getLecture().getCourse().hashCode();
+        return new PyrisLectureUnitWebhookDTO(false, artemisBaseUrl, "", lectureUnitId, "", lectureId, "", courseId, "", "");
+    }
+
+    /**
+     * send the updated / created attachment to Pyris for ingestion if autoLecturesUpdate is enabled
+     *
+     * @param courseId               Id of the course where the attachment is added
+     * @param newAttachmentUnits     the new attachment Units to be sent to pyris for ingestion
+     * @param irisSettingsRepository the repository to get the course settings from
+     * @return true if the units were sent to pyris
+     */
+    public boolean autoUpdateAttachmentUnitsInPyris(IrisSettingsRepository irisSettingsRepository, Long courseId, List<AttachmentUnit> newAttachmentUnits) {
+        try {
+            IrisCourseSettings courseSettings = irisSettingsRepository.findCourseSettings(courseId).orElseThrow(() -> new NoSuchElementException("Course settings not found"));
+            if (courseSettings.getIrisLectureIngestionSettings() != null && courseSettings.getIrisLectureIngestionSettings().isEnabled()
+                    && Boolean.TRUE.equals(courseSettings.getIrisLectureIngestionSettings().getAutoIngestOnLectureAttachmentUpload())) {
+                return addLectureToPyrisDB(newAttachmentUnits) != null;
+            }
+        }
+        catch (NoSuchElementException e) {
```

Comment:
Don't catch an exception that you just threw in the same method. Please use an if-statement with `isPresent`/`isEmpty` check